### PR TITLE
Document template CMake files

### LINKED ISSUE #93930: CMakeList.txt for windows contains incompatible CMake setting
The template for `CMakeList.txt` for the Windows platform contains this line:
https://github.com/flutter/flutter/blob/1c36271b058ab798463f4348f16b7cacb7edb5ca/packages/flutter_tools/templates/app_shared/windows.tmpl/CMakeLists.txt.tmpl#L8

`CMAKE_INSTALL_RPATH` is not applicable to Windows, since the platform has no concept of rpaths. It's harmless to set this variable, but might confuse developers.

### LINKED ISSUE #94866: Document the Windows and Linux CMake template files
Currently the files are largely undocumented; in a recent discussion of templates Hixie pointed out that when possible our template files should be heavily documented to allow someone who has little to no familiarity with, in this case, CMake and the project structure to understand how to edit it.

### PR DESCRIPTION
**Pull request by stuartmorgan-g**
Adds comments throughout the CMake files in the Windows and Linux
templates to make them easier to understand. This helps guide
developers, especially those who may not be familiar with CMake, as to
what the various parts do and which should or should not be edited in
general.

Fixes https://github.com/flutter/flutter/issues/94866
Fixes https://github.com/flutter/flutter/issues/93930

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/templates/app_shared/windows.tmpl/CMakeLists.txt.tmpl [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
+# The name of the executable created for the application. Change this to change
+# the on-disk name of your application.
 set(BINARY_NAME "{{projectName}}")
 
+# Explicitly opt in to modern CMake behaviors to avoid warnings with recent
+# versions of CMake.
 cmake_policy(SET CMP0063 NEW)
 
-set(CMAKE_INSTALL_RPATH "$ORIGIN/lib")
```
Comment (stuartmorgan) [inline packages/flutter_tools/templates/app_shared/windows.tmpl/CMakeLists.txt.tmpl:8]: This is the only non-comment change; see https://github.com/flutter/flutter/issues/93930

File: packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl [hunk 2]
```diff
@@ -1,11 +1,18 @@
+# Project-level configuration.
 cmake_minimum_required(VERSION 3.10)
 project(runner LANGUAGES CXX)
 
+# The name of the executable created for the application. Change this to change
+# the on-disk name of your application.
 set(BINARY_NAME "{{projectName}}")
+# The unique GTK application identifier for this application.
```
Comment (cbracken) [inline packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl:8]: I wonder if it's worth referencing the relevant GTK docs. One issue is that I can't actually _find_ the GTK docs for this other than a bunch of references in Gnome docs. Gnome is built on GTK but I'm not sure whether or not they impose any additional requirements on application IDs that GTK does not. :/

https://developer-old.gnome.org/ChooseApplicationID/

File: packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 set(APPLICATION_ID "{{linuxIdentifier}}")
 
+# Explicitly opt in to modern CMake behaviors to avoid warnings with recent
+# versions of CMake.
 cmake_policy(SET CMP0063 NEW)
 
+# Ensure that bundled libraries are loaded from the correct location.
```
Comment (cbracken) [inline packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl:15]: Maybe more precisely (as I understand it):
```
# Ensure that bundled libraries are loaded from the lib/ directory relative to the binary.
```

**Review by cbracken — APPROVED** [review]
lgtm modulo a couple nits.

File: packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl (hunk 3 above)
Comment (stuartmorgan) [reply packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl]: Done.

**Pull request merged**